src/MongoDB.Driver.Core/Core/Compression/Snappy/Native/SnappyNativeMethods.cs

```diff
@@ -0,0 +1,96 @@
+﻿/* Copyright 2019–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using MongoDB.Driver.Core.Compression.Native;
+using MongoDB.Driver.Core.NativeLibraryLoader;
+
+namespace MongoDB.Driver.Core.Compression.Snappy.Native
+{
+    internal static class SnappyNativeMethods
```

Comment:
Let's call this class `Snappy64NativeMethods`.

I would like to see this class stripped to a single responsibility: invoking the native methods (let's say loading the native method is part of the same responsibility).

See my branch for a prototype.

Reply:
Let's layer two more classes on top of this to divide and conquer the work:

SnappyAdapter: to convert byte arrays to IntPtr
Snappy64Adapter: to convert int to ulong (int being the preferred C# type and ulong being what the native library uses)

See my branch for a prototype.

Reply:
OK. We can consolidate some of the layers.